rules/no-hooks.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+module.exports = {
+  meta: {
+    docs: {
+      url:
+        'https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-hooks.md',
+    },
+  },
+  schema: [
+    {
+      type: 'object',
+      properties: {
+        allowed: {
+          type: 'array',
+          contains: ['beforeAll', 'beforeEach', 'afterAll', 'afterEach'],
+        },
+      },
+      additionalProperties: false,
+    },
+  ],
+  create(context) {
+    const testHookNames = Object.assign(Object.create(null), {
+      beforeAll: true,
+      beforeEach: true,
+      afterAll: true,
+      afterEach: true,
+    });
+
+    const allowed = (context.options[0] || { allowed: [] }).allowed.reduce(
+      (hashMap, value) => {
+        hashMap[value] = true;
+        return hashMap;
+      },
+      Object.create(null)
+    );
+
+    const isHook = node => testHookNames[node.callee.name];
+    const isWhitelisted = node => allowed[node.callee.name];
+
+    return {
+      CallExpression(node) {
+        if (isHook(node) && !isWhitelisted(node)) {
+          context.report({
+            node,
+            message: 'Unexpected setup/teardown hook',
```

Comment:
Should the message including the hook name (e.g. beforeEach), or is this sufficient?

Reply:
I'm going to include the hook name in the error message. 👍